sled-agent/src/profile.rs

```diff
@@ -35,6 +35,13 @@ impl ProfileBuilder {
 
         let profile_path = installed_zone.site_profile_xml_path();
 
+        // Create the xml profile directory if it doesn't already exist.
+        // All self assembled zones already have this directory;
+        // this is mostly for our testing.
+        let mut path = profile_path.clone();
+        path.pop();
+        tokio::fs::create_dir_all(path).await?;
+
```

Comment:
I don't like this at all, but even though [`fake_install()`](https://github.com/oxidecomputer/omicron/blob/main/illumos-utils/src/running_zone.rs#L1301-L1304) creates the directory, the tests kept reporting the directory didn't exist :(

@smklein and/or @citrus-it , if you have a better suggestion on how to go on about this I'd love to hear it!

Reply:
Never mind, fixed in https://github.com/oxidecomputer/omicron/pull/5168/commits/6718070742f09b9a5c343c20e6a48e05f32916b5 :)